Mqtt/BaseMessageHandler.cs

```diff
@@ -7,11 +7,11 @@ namespace MRS.Mqtt;
 
 public abstract class BaseMessageHandler
 {
-    protected class MessageWrapper
+    protected struct MessageWrapper
     {
-        [JsonPropertyName("username")] public string Username;
-        [JsonPropertyName("source")] public string Source;
-        [JsonPropertyName("payload")] public Object Payload;
+        [JsonPropertyName("username")] public string username { get; set; }
+        [JsonPropertyName("source")] public string source { get; set; }
+        [JsonPropertyName("payload")] public object payload { get; set; }
     }
 
     protected MqttApplicationMessageReceivedEventArgs _messageEvent;
```

Comment:
Scan is correct, you need to either declare that the fields can be nullable (and handle null values where necessary), or ensure they have a value when created